File: src/org/genyris/test/interp/TestUtilities.java
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.test.interp;

import java.io.StringWriter;

import org.genyris.core.Exp;
import org.genyris.exception.GenyrisException;
import org.genyris.format.BasicFormatter;
import org.genyris.format.Formatter;
import org.genyris.interp.Interpreter;
import org.genyris.io.Parser;

public class TestUtilities {

    public Interpreter _interpreter;

    public TestUtilities() throws GenyrisException {
        _interpreter = new Interpreter();
        _interpreter.init(false);
    }


   public String eval(String script) throws GenyrisException {
        Exp expression = Parser.parseSingleExpressionFromString(_interpreter.getSymbolTable(), script);
        Exp result = _interpreter.evalInGlobalEnvironment(expression);

        StringWriter out = new StringWriter();
        Formatter formatter = new BasicFormatter(out);
        result.acceptVisitor(formatter);
        return out.getBuffer().toString();
    }


}

File: src/org/genyris/test/interp/TestUtilities.kt
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.test.interp

import org.genyris.core.Exp
import org.genyris.exception.GenyrisException
import org.genyris.format.BasicFormatter
import org.genyris.format.Formatter
import org.genyris.interp.Interpreter
import org.genyris.io.Parser
import java.io.StringWriter

class TestUtilities {
    var _interpreter: Interpreter

    init {
        _interpreter = Interpreter()
        _interpreter.init(false, "TestUtilities")
    }


    @kotlin.Throws(GenyrisException::class)
    fun eval(script: String?): String {
        val expression: Exp = Parser.Companion.parseSingleExpressionFromString(_interpreter.getSymbolTable(), script)
        val result = _interpreter.evalInGlobalEnvironment(expression)

        return renderExp(result, false)
    }

    companion object {
        @kotlin.Throws(GenyrisException::class)
        fun renderExp(X: Exp, expandAbbreviation: Boolean): String {
            val out = StringWriter()
            val formatter: Formatter = BasicFormatter(out, expandAbbreviation)
            X.acceptVisitor(formatter)
            return out.getBuffer().toString()
        }
    }
}
